src/firewall/nft.rs

```diff
@@ -1058,6 +1206,26 @@ fn get_rule_matcher_jump_to(jump_target: String) -> Box<dyn Fn(&schema::Rule) ->
     })
 }
 
+/// Make a closure that matches any rule that tests for a match to a given bridge interface.
+fn get_rule_matcher_bridge(bridge: String) -> Box<dyn Fn(&schema::Rule) -> bool> {
+    Box::new(move |r: &schema::Rule| -> bool {
```

Comment:
we do not need the box here and we can also get rid of the bridge String with the use of a lifetime.
```diff
diff --git a/src/firewall/nft.rs b/src/firewall/nft.rs
index b123a2a..9c75380 100644
--- a/src/firewall/nft.rs
+++ b/src/firewall/nft.rs
@@ -246,7 +246,7 @@ impl firewall::FirewallDriver for Nftables {
             ));
         }
 
-        let match_our_bridge = get_rule_matcher_bridge(network_setup.bridge_name.clone());
+        let match_our_bridge = get_rule_matcher_bridge(&network_setup.bridge_name);
 
         // If and only if isolation is enabled: add isolation chains.
         // Some isolation rules are shared. Other rules are specific to one type
@@ -506,7 +506,7 @@ impl firewall::FirewallDriver for Nftables {
             }
         }
 
-        let match_our_bridge = get_rule_matcher_bridge(tear.config.bridge_name.clone());
+        let match_our_bridge = get_rule_matcher_bridge(&tear.config.bridge_name);
 
         let mut isolation_rules: Vec<schema::Rule> = Vec::new();
         isolation_rules.append(&mut get_matching_rules_in_chain(
@@ -1207,13 +1207,13 @@ fn get_rule_matcher_jump_to(jump_target: String) -> Box<dyn Fn(&schema::Rule) ->
 }
 
 /// Make a closure that matches any rule that tests for a match to a given bridge interface.
-fn get_rule_matcher_bridge(bridge: String) -> Box<dyn Fn(&schema::Rule) -> bool> {
-    Box::new(move |r: &schema::Rule| -> bool {
+fn get_rule_matcher_bridge(bridge: &str) -> impl '_ + Fn(&schema::Rule) -> bool {
+    move |r: &schema::Rule| -> bool {
         for statement in &r.expr {
             match statement {
                 stmt::Statement::Match(m) => match &m.right {
                     expr::Expression::String(s) => {
-                        if *s == bridge {
+                        if s == bridge {
                             return true;
                         }
                     }
@@ -1223,7 +1223,7 @@ fn get_rule_matcher_bridge(bridge: String) -> Box<dyn Fn(&schema::Rule) -> bool>
             }
         }
         false
-    })
+    }
 }
 
 /// Find all rules in the given chain which match the given closure (true == include).
```

Reply:
Changes pushed, TY